### PR TITLE
Issues a warning when constructor invocation type is a subtype of the constructor declaration type; Fixes #2163

### LINKED ISSUE #2163: Qualified type of constructor invocations
The qualified type of the constructor invocation `new MyClass(22)` depends on three annotations:

```
class @A MyClass {
  @B MyClass(int arg) { ... }
}

class Client {
  ... new @C MyClass(22) ...
}
```

These three annotations must have the following relationship:
```
  @C <: @B <: @A
```
If any of these relationships is violated, that is an error.

The qualified type of the constructor invocation is `@C`.  (That is also equal to the greatest lower bound of `@A`, `@B`, and `@C`.)

If `@C` is omitted, then the type of `new MyClass(22)` is `@B MyClass`.
If `@C` and `@B` are omitted, then the type of `new MyClass(22)` is `@A MyClass`, just like any other use of `MyClass` without an explicit annotation.
If `@B` is omitted, then the result type of the constructor ` MyClass(int arg) { ... }` and the type  of `this` inside the constructor are `@A MyClass`.  This is the same for any other use of `MyClass` without an explicit annotation.

Currently, it seems that the Checker Framework may sometimes silently ignore the annotation `@C` (see https://github.com/typetools/checker-framework/issues/2159).
The Checker Framework should never ignore any annotation.  If an annotation is inconsistent with other annotations, the Checker Framework should issue an error.

### PR DESCRIPTION
**Pull request by t-rasmud**
Fixes #2163.

### CODE REVIEW

File: checker/tests/interning/Creation.java [hunk 1]
```diff
@@ -12,6 +12,7 @@ public void test() {
         // :: error: (assignment.type.incompatible)
         @Interned Foo f = new Foo(); // error
         Foo g = new Foo(); // valid
+        // :: warning: (cast.unsafe)
```
[the remark this answers is not in the document]
Comment (t-rasmud) [reply checker/tests/interning/Creation.java]: This warning should be produced whenever the type at constructor invocation is a subtype of the constructor declaration type (More details [here](https://github.com/typetools/checker-framework/issues/2163#issuecomment-446326482)). I have tried it with the property key checker which is simple enough.

[the remark this answers is not in the document]
Comment (mernst) [reply checker/tests/interning/Creation.java]: In other words:  yes, you are right that a lot of warnings are added.  However, they are all desirable warnings:  they were missing before.

File: checker/src/test/java/tests/PropertyKeyTest.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package tests;
+
+import java.io.File;
+import java.util.List;
+import org.checkerframework.framework.test.CheckerFrameworkPerDirectoryTest;
+import org.junit.runners.Parameterized.Parameters;
+
+public class PropertyKeyTest extends CheckerFrameworkPerDirectoryTest {
```
[the remark this answers is not in the document]
Comment (t-rasmud) [reply checker/src/test/java/tests/PropertyKeyTest.java]: Thanks @wmdietl . I have moved this test to the i18n directory.

**Comment by mernst** [comment]
@wmdietl FYI, the substantive changes are just two statements at https://github.com/typetools/checker-framework/pull/2228/files#diff-31033326203125d5a0af0323436d39f4R2433 .

**Pull request merged**